department-of-veterans-affairs/va.gov-cms
# VACMS-12394: Prevent links with VA.gov domains as first path segment.

## Description

Closes #12394.

## QA steps

As any user with the ability to edit content:
1. Edit a node, e.g. [VAMC System Health Service Whole health at VA Cincinnati health care](https://cms-wpuulvbzdyrcqodsktt7i3ppei6scpm3.ci.cms.va.gov/node/24426/edit)
2. Insert a relative link containing a VA.gov domain into a rich text field.  For instance: `<a href="/www.va.gov/BUILD.txt">Example Link!</a>`.
3. Attempt to save the document.  You should see:
  - [x] An error message at the top of the screen:
![Screen Shot 2023-02-01 at 11 02 56 AM](https://user-images.githubusercontent.com/1318579/216098698-ed3ec7ae-89a3-4a4b-b546-90dad9b1a2b2.png)

  - [ ] validate error message link jumps to the error.

  - [x] An error message at the field:
![Screen Shot 2023-02-01 at 11 11 38 AM](https://user-images.githubusercontent.com/1318579/216098714-9ff6947e-2d4d-4a8a-af63-96546f95498a.png)

  - [x] An error message at the bottom of the field:
![Screen Shot 2023-02-01 at 11 11 43 AM](https://user-images.githubusercontent.com/1318579/216098741-e762b894-5a0a-4bcf-9835-81aef6804449.png)

4. Edit a media image item, e.g. [August2022.jpg](https://cms-wpuulvbzdyrcqodsktt7i3ppei6scpm3.ci.cms.va.gov/media/image/16769)

5.  Insert a relative link containing a URL into a plain text field. For instance, `/www.va.gov/BUILD.txt`.

6. Attempt to save the media image item.  You should see:
  - [x] An error message at the top of the screen:
![Screen Shot 2023-02-01 at 11 12 24 AM](https://user-images.githubusercontent.com/1318579/216098934-69352628-70d8-4698-a6a9-1186bd2452a5.png)

  - [ ] validate error message link jumps to the error.

  - [x] An error message at the field:
![Screen Shot 2023-02-01 at 11 12 28 AM](https://user-images.githubusercontent.com/1318579/216098947-1ec1a206-c91c-4c0c-822a-1f909316bc4e.png)

  - [x] An error message at the bottom of the field:
![Screen Shot 2023-02-01 at 11 12 32 AM](https://user-images.githubusercontent.com/1318579/216098963-d93915a9-2f2c-40f2-9fcf-82c0c6c32c8d.png)

### Definition of Done
